### ISSUE TITLE
PANDA dataset

### ISSUE THREAD

**Issue by ahmedemadedlin** [prov-gigapath/prov-gigapath#23]
Hello, 
i wanted to ask about the output from the classification layer
 I used a sample from the PANDA dataset it is supposed to output array of 6 the number of classed but I'm only getting array of 2

**Comment by HanwenXuTHU** [comment]
Hi, can you post the codes you're testing? I will take a look.

**Comment by ahmedemadedlin** [comment]
import timm
from PIL import Image
from torchvision import transforms
import torch
import gigapath.slide_encoder as slide_encoder
import gigapath.classification_head as cls

tile_encoder = timm.create_model("hf_hub:prov-gigapath/prov-gigapath", pretrained=True)

transform = transforms.Compose(
    [
        transforms.Resize(256, interpolation=transforms.InterpolationMode.BICUBIC),
        transforms.CenterCrop(224),
        transforms.ToTensor(),
        transforms.Normalize(mean=(0.485, 0.456, 0.406), std=(0.229, 0.224, 0.225)),
    ]
)
img_path = "images/prov_normal_000_1.png"
img = Image.open(img_path).convert("RGB")
sample_input = transform(img).unsqueeze(0)

tile_encoder.eval()
tile_embeddings = []
# tile_embed = None
with torch.no_grad():
    for i in range(2):
        tile_embed = tile_encoder(sample_input).squeeze()
        tile_embeddings.append(tile_embed)

tile_embeddings = torch.stack(tile_embeddings)
tile_embeddings = tile_embeddings.unsqueeze(0).to('cuda')

coords = torch.tensor([2,2])
coords = coords.to('cuda')

slide_encoder = slide_encoder.create_model("hf_hub:prov-gigapath/prov-gigapath", "gigapath_slide_enc12l768d", 1536)
slide_encoder.cuda()

slide_encoder.eval()

with torch.no_grad():
    output = slide_encoder(tile_embeddings,coords)

cls_model = cls.ClassificationHead(input_dim=1536,latent_dim=768,feat_layer='11')
cls_model = cls_model.cuda()
with torch.no_grad():
    result = cls_model(images=tile_embeddings,coords=coords)
    print(result)

**Comment by ahmedemadedlin** [comment]
Thanks, I solved it 
cls_model = cls.ClassificationHead(input_dim=1536,latent_dim=768,feat_layer='11',n_classes=6)

**Comment by ahmedemadedlin** [comment]
Thanks, I solved it
cls_model = cls.ClassificationHead(input_dim=1536,latent_dim=768,feat_layer='11',n_classes=6)

**Comment by HanwenXuTHU** [comment]
Great to hear you have figured it out! Let me know if you have any other questions!

**Comment by ahmedemadedlin** [comment]
sure, Thanks
